### PR TITLE
Feature/nft operations

### PR DESCRIPTION
**Pull request by lukas1**

### CODE REVIEW

File: Sources/DCoreKit/Model/Operations/NftCreateOperation.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                options.description.count <= DCore.Constant.uiaDescriptionMaxChars,
+                "description cannot be longer than \(DCore.Constant.uiaDescriptionMaxChars) chars"
+            )
+            precondition(options.maxSupply <= 0xFFFFFFFF)
+        }
+    }
+    public var definitions: [NftDataType] {
+        willSet {
+            precondition(
+                definitions.allSatisfy { $0.modifiable == NftDataType.ModifiableBy.nobody || !$0.name.isNil() },
```
Comment (fl3xman) [inline Sources/DCoreKit/Model/Operations/NftCreateOperation.swift:21]: $0.modifiable == NftDataType.ModifiableBy.nobody || !$0.name.isNil()  is this condition correct?

Comment (lukas1) [reply Sources/DCoreKit/Model/Operations/NftCreateOperation.swift:21]: yes. If it's modifiable, name is required

File: Sources/DCoreKit/Model/NftDataType.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            switch self {
+            case .nobody: return 0
+            case .issuer: return 1
+            case .owner: return 2
+            case .both: return 3
+            }
+        }
+    }
+
+    public enum DataType: String, Codable {
```
Comment (fl3xman) [inline Sources/DCoreKit/Model/NftDataType.swift]: NftDataType.DataType is little ambiguous/confusing, can u suggest different name for DataType.

Comment (fl3xman) [reply Sources/DCoreKit/Model/NftDataType.swift]: @lukas1 what if we call it `TypeLiteral`?

**Review by fl3xman — APPROVED** [review]

**Pull request merged**